### ISSUE TITLE
AttributeError: 'float' object has no attribute 'split'

### ISSUE THREAD

**Issue by herber4** [splicebox/Jutils#1]
Can anyone help me with this error?

herber4@245 Jutils % python3 jutils.py heatmap --tsv-file ../j_out/rmats_JC_results.tsv --meta-file meta.txt --out-dir ../GTF 
Traceback (most recent call last):
  File "/Users/herber4/Desktop/Jutils/jutils.py", line 145, in <module>
    main()
  File "/Users/herber4/Desktop/Jutils/jutils.py", line 138, in main
    run_heatmap_module(args, parser_dict)
  File "/Users/herber4/Desktop/Jutils/jutils.py", line 106, in run_heatmap_module
    plot_heatmap(Path(args.tsv_file), Path(args.meta_file), Path(args.out_dir), args.p_value,
  File "/Users/herber4/Desktop/Jutils/heatmap_utils.py", line 60, in plot_heatmap
    data_df['GeneName'] = data_df['GeneName'].apply(truncate_gene_name)
  File "/Users/herber4/Library/Python/3.9/lib/python/site-packages/pandas/core/series.py", line 4357, in apply
    return SeriesApply(self, func, convert_dtype, args, kwargs).apply()
  File "/Users/herber4/Library/Python/3.9/lib/python/site-packages/pandas/core/apply.py", line 1043, in apply
    return self.apply_standard()
  File "/Users/herber4/Library/Python/3.9/lib/python/site-packages/pandas/core/apply.py", line 1099, in apply_standard
    mapped = lib.map_infer(
  File "pandas/_libs/lib.pyx", line 2859, in pandas._libs.lib.map_infer
  File "/Users/herber4/Desktop/Jutils/heatmap_utils.py", line 80, in truncate_gene_name
    gene_names = string.split(',')
AttributeError: 'float' object has no attribute 'split'
herber4@245 Jutils % python3 jutils.py heatmap --tsv-file ../j_out/rmats_JC_results.tsv --meta-file meta.txt --out-dir ../j_out
Traceback (most recent call last):
  File "/Users/herber4/Desktop/Jutils/jutils.py", line 145, in <module>
    main()
  File "/Users/herber4/Desktop/Jutils/jutils.py", line 138, in main
    run_heatmap_module(args, parser_dict)
  File "/Users/herber4/Desktop/Jutils/jutils.py", line 106, in run_heatmap_module
    plot_heatmap(Path(args.tsv_file), Path(args.meta_file), Path(args.out_dir), args.p_value,
  File "/Users/herber4/Desktop/Jutils/heatmap_utils.py", line 60, in plot_heatmap
    data_df['GeneName'] = data_df['GeneName'].apply(truncate_gene_name)
  File "/Users/herber4/Library/Python/3.9/lib/python/site-packages/pandas/core/series.py", line 4357, in apply
    return SeriesApply(self, func, convert_dtype, args, kwargs).apply()
  File "/Users/herber4/Library/Python/3.9/lib/python/site-packages/pandas/core/apply.py", line 1043, in apply
    return self.apply_standard()
  File "/Users/herber4/Library/Python/3.9/lib/python/site-packages/pandas/core/apply.py", line 1099, in apply_standard
    mapped = lib.map_infer(
  File "pandas/_libs/lib.pyx", line 2859, in pandas._libs.lib.map_infer
  File "/Users/herber4/Desktop/Jutils/heatmap_utils.py", line 80, in truncate_gene_name
    gene_names = string.split(',')
AttributeError: 'float' object has no attribute 'split'

**Comment by Guangyu-Yang** [comment]
Hi herber4,
Sorry for my late reply, this problem may associate with the gene names in the rmats_JC_results.tsv file. One reason might be  that the gene names only contain digit, so the parser recognize them as float other than string. Do you use a custom tsv or it's generated by "python3 jutils.py convert-results"? To help us debug the error, can you share some gene names from the rmats_JC_results.tsv?

**Comment by LPChaumont** [comment]
Hi Guangyu-Yang,

I tried your tool today and got the same error. The reason was that some rows in rmats_JC_results.tsv had no values in the GeneName column. This was because not all gene IDs in the rMATS results had a gene name. When I removed the rows with missing gene names, the error message disappeared. 

To solve this inconvenience, the command `jutils.py convert-results` could use the value in the column "GeneID" when there is no gene name in "geneSymbol". But I think it would probably be more safe to keep the gene IDs and the gene names in rmats_JC_results.tsv because one gene name can be associated to multiple gene IDs.

Cheers
